client/src/App.js

```diff
@@ -1,28 +1,59 @@
-import React from "react";
+import React, { useContext, useEffect } from "react";
 import { MuiThemeProvider } from "@material-ui/core";
 import { BrowserRouter, Route, Switch } from "react-router-dom";
 import { theme } from "./themes/theme";
+import Navbar from './components/layout/Navbar'
 import Signup from "./pages/signup";
 import Signin from "./pages/signin";
-import Dashboard from "./pages/Dashboard";
-import Home from "./pages/TempHome";
-import Profile from "./pages/Profile";
+import Dashboard from './pages/Dashboard';
+import FAQ from './pages/FAQ';
+import Blog from './pages/Blog';
+import Profile from './pages/Profile';
+import Home from "./pages/HomePage";
+import axios from 'axios';
+import { UserProvider } from './context/UserContext';
+import { AuthContext } from './context/AuthContext';
 
 import "./App.css";
 
 function App() {
-  return (
+  const [isAuthenticated, setIsAuthenticated] = useContext(AuthContext);
+
+  useEffect(() => {
+    axios.get('/api/JWT')
+      .then(() => {
+        setIsAuthenticated(true)
+      })
+      .catch((err) => {
+        console.log(err)
+        setIsAuthenticated(false)
+      })
+  })
```

Comment:
missing dependency here so this will run every time it re-renders

Reply:
Added!